spectrum/input.py

```diff
@@ -183,14 +184,44 @@ def _choose_submit(self, wrapped_form, name, value=None):
                 continue
             del inp['name']
 
-    def _assert_html_response(self, response):
-        assert response.status_code == 200, "Response from saving the from was expected to be 200 from the listing page, but it was %s\nBody: %s" % (response.status_code, response.content)
+def _assert_html_response(response):
+    assert response.status_code == 200, "Response from saving the from was expected to be 200 from the listing page, but it was %s\nBody: %s" % (response.status_code, response.content)
 
 def _journal_cms_page_title(soup):
     # <h1 class="js-quickedit-page-title title page-title"><span data-quickedit-field-id="node/1709/title/en/full" class="field field--name-title field--type-string field--label-hidden">Spectrum blog article: jvsfz4oj9vz9hk239fbpq4fbjc9yoh</span></h1>
     #<h1 class="js-quickedit-page-title title page-title">alfred</h1>
     return soup.find("h1", {"class": "page-title"}).text.strip()
 
+class Journal:
+    def __init__(self, host):
+        self._host = host
+
+    def session(self):
+        browser = mechanicalsoup.Browser()
+        return JournalSession(self._host, browser)
+
+class JournalSession:
+    def __init__(self, host, browser):
+        self._host = host
+        self._browser = browser
+
+    def login(self):
+        self._enable_feature_flag()
+
+        login_url = "%s/log-in" % self._host
+        # should be automatically redirected back by simulator
+        logged_in_page = self._browser.get(login_url)
+        _assert_html_response(logged_in_page)
+
+        pattern = re.compile(r'Log out')
+        logout_link = logged_in_page.soup.find("a", text=pattern)
+        assert logout_link.get('href'), '/log-out'
+
+    def _enable_feature_flag(self):
```

Comment:
`_enable_can_authenticate_feature`?

Reply:
I thought it was in general enabling all hidden features, since it's named generically, but can rename

Reply:
Each feature would be done on a case-by-case basis. We probably want individual enablers if we do this technique again. (Maybe `?enable=can_authenticate` would have been better.)